### DISCUSSION TITLE
[CRUD service] Possible enhancement on state transition request

### DISCUSSION THREAD

**Discussion by SimoneBergonzi** [mia-platform/community#218] [Q&A]
Hi!

Reading the documentation I get that when performing a /state request you can can only specify the new state in which you want to set the document and nothing else.
Furthermore, the only API which can update the __STATE__ of a document is the /state API.

This means that, if I want to update some other properties when changing the __STATE__ of a document, I have to perform 2 separate calls.

Is there anybody who encountered the same scenario? Could it be useful to open the possibility of modifying other fields while updating the __STATE__ of a document?

**Comment by danibix95** [comment]
Hi, thanks for the feedback. Currently `__STATE__` is treated as internal field, as shown [here](https://github.com/mia-platform/crud-service/blob/main/lib/httpInterface.js#L923-L929). Therefore it is not possible to update the `__STATE__` field with a standard PATCH operation.

We need to understand whether we should allow modifying the `__STATE__` as you suggested. Could you open an issue on the [CRUD Service repository](https://github.com/mia-platform/crud-service) with this feature request?

FYI @TheRusCompass
